nixos/modules/services/misc/sourcehut/default.nix

```diff
@@ -56,6 +67,13 @@ in
       '';
     };
 
+    hostName = mkOption {
```

Comment:
Would it be better to just call it `fqdn`?

Reply:
Yeah, it can be fqdn too. not sure what to call it besides `subdomainName` or something

Reply:
originBase?

Reply:
In response to the structure of the settings: Right now the `settings` reflects the `config.ini` directly and allows us to use the [RFC 42-style](https://github.com/NixOS/rfcs/blob/45b76f20add8d7e0e1d0bc0eed9357b067c899a6/rfcs/0042-config-option.md) approach for configuration. It can be made a bit cleaner by auto-appending "sr.ht" or building the configs, but there are exceptions and using the INI generator is pretty nice.

As to the array of services, I think the original PR was using an approach along the lines [plugins](https://search.nixos.org/options?channel=20.09&show=services.tt-rss.plugins&from=0&size=50&sort=relevance&query=plugins). I agree that it isn't the cleanest right now. I do have a section of config options for each service, but they seem to conflict or duplicate what is in `settings`.

I'd like to get some reasonable baseline merged and improve on this, not sure what that will be though yet. I want to add better documentation of the defaults, not sure exactly the best way to do that with Freeform modules, this is my first module in this style.

Updated branch pushed.

Reply:
Understood. Thanks for the response and the work so far! It's looking great.